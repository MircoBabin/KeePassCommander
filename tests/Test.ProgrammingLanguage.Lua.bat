@echo off
    setlocal
    call "%~dp0Utils\Utils.test-setup.bat" "%~f0"

    chcp 65001 > nul 2>&1
    "%~dp0..\assets\lua\lua54.exe" "%~dp0Test.ProgrammingLanguage.Lua.lua" "%KeePassCommandExe%" > "%KeePassCommand_TestOutput%"
    
    call "%~dp0Utils\Utils.diff.bat"
    if errorlevel 1 exit /b 1
    exit /b 0