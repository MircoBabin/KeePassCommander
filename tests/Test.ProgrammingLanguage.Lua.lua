local function printTable(Table, Ident)
    -- collect the keys
    local keys = {}
    for key in pairs(Table) do
        keys[#keys+1] = key
    end

    -- sort the keys
    table.sort(keys)

    -- output the table based on sorted keys
    for i=1,#keys do
        local key = keys[i]
        if type(Table[key]) == 'table' then
            print(Ident..'"'..key..'"= {')
            printTable(Table[key], Ident..'    ')
            print(Ident..'}')
        else
            print(Ident..'"'..key..'"='..Table[key])
        end
    end
end

-- require KeePassEntry.lua containing class KeePassEntry
package.path = './../src/ProgrammingLanguagesConnectors/KeePassEntry.lua;' .. package.path
local KeePassEntry = require("KeePassEntry")

-- set the Options
local EntryName = "Sample Entry"
local Options = {}
Options.KeePassCommandExe = arg[1]
Options.FieldNames = { "extra field 1", "extra password 1" }
Options.AttachmentNames = { "example_attachment.txt" }

-- retrieve KeePass entry
local entry = KeePassEntry.get(EntryName, Options)
if string.len(entry.title) == 0 then
    print('KeePass is not started')
    print('Has KeePassCommander.dll been copied to the directory containing KeePass.exe ?')

    return 2
end

-- ordered output
printTable(entry, '')

return 0
