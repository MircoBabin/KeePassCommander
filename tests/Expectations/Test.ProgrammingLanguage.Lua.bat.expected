"attachments"= {
    "example_attachment.txt"=This is an attachment.

With three lines in it.
}
"fields"= {
    "extra field 1"=Some extra data
    "extra password 1"=Some extra password
}
"notes"=Notes Één €ûrò
"password"=Password
"title"=Sample Entry
"url"=https://keepass.info/
"urlhost"=keepass.info
"urlpath"=/
"urlport"=443
"urlscheme"=https
"username"=User Name
